src/main/java/com/testjava/metrics_platform/TestJavaEventSender.java

```diff
@@ -18,38 +23,40 @@
 import com.google.gson.Gson;
 
 public class TestJavaEventSender implements EventSender {
+
+    private final Gson gson = new Gson();
+
     @Override
-    public void sendEvents(String baseUri, Collection<Event> events) throws MalformedURLException {
+    public void sendEvents(String baseUri, Collection<Event> events) throws IOException {
         URL url = isValid(baseUri) ? new URL(baseUri) : new URL(DestinationEventService.LOCAL.getBaseUri());
 
+        HttpURLConnection connection = null;
         try {
-            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection = (HttpURLConnection) url.openConnection();
             connection.setRequestMethod("POST");
             connection.setRequestProperty("Accept", "application/json");
             connection.setRequestProperty("Content-Type", "application/json");
             connection.setDoOutput(true);
+            connection.connect();
 
-            for (Event event : events) {
-                String payload = new Gson().toJson(event);
-                System.out.println(payload);
-                byte[] postData = payload.getBytes(StandardCharsets.UTF_8);
-                try (DataOutputStream dos = new DataOutputStream(connection.getOutputStream())) {
-                    dos.write(postData, 0, postData.length);
-                    System.out.println("Posted data to local event logging!");
+            try (Writer writer = new BufferedWriter(new OutputStreamWriter(connection.getOutputStream(), UTF_8))) {
+                for (Event event : events) {
+                    gson.toJson(event, writer);
                 }
-                try (BufferedReader br = new BufferedReader(
-                        new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8))) {
-                    StringBuilder content = new StringBuilder();
-                    String inputLine;
-                    while ((inputLine = br.readLine()) != null) {
-                        content.append(inputLine.trim());
-                    }
-                    System.out.println(content);
+                System.out.println("Posted data to local event logging!");
+            }
+
+            int status = connection.getResponseCode();
+            if (status == HttpURLConnection.HTTP_OK) {
+                try (BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream(), UTF_8))) {
+                    br.lines().forEach(System.out::print);
+                    System.out.println();
                 }
+            } else {
+                throw new IOException("Error sending events, status code=" + status);
             }
-            connection.disconnect();
-        } catch (IOException e) {
-            e.printStackTrace();
+        } finally {
+            if (connection != null) connection.disconnect();
```

Comment:
disconnect() should be called in a finally block. It's a shame that HttpUrlConnection doesn't implement Closeable, which would allow to use try-with-resources.